langchain4j-core/src/main/java/dev/langchain4j/model/chat/request/ChatParameters.java

```diff
@@ -0,0 +1,36 @@
+package dev.langchain4j.model.chat.request;
+
+import dev.langchain4j.Experimental;
+import dev.langchain4j.agent.tool.ToolSpecification;
+
+import java.util.List;
+
+@Experimental
+public interface ChatParameters {
```

Comment:
Related to what I wrote regarding the introduction of that `Utils.ifNotNull` method, maybe we may want to clarify which values are mandatory and which ones can be empty. For instance I guess that you used the boxed types `Double` and `Integer` instead of the corresponding primitive ones because you use `null` when they are undefined. Wouldn't be better to declare the non mandatory fields as `Optional` so they could be found at a glance?

Slightly related to this should we return some `default` value when such a meaningful default exists? For instance should the `temperature()` return `0.0` by default or the `maxOutputTokens()` return `Integer.MAX_VALUE`?

Reply:
> maybe we may want to clarify which values are mandatory and which ones can be empty.

I guess the required/optional will depend on when `ChatParameters` are specified. `ChatParameters` can be specified when creating the model (per-model) and/or when calling it (per-request).
For example, `modelName` is mandatory in most of the cases when calling LLM API, but if it was specified when creating the model, it does not need to be specified again when calling it. And vice versa, if it is optional when creating the model, then it is mandatory when calling it.

> For instance I guess that you used the boxed types Double and Integer instead of the corresponding primitive ones because you use null when they are undefined.

Yes

> Wouldn't be better to declare the non mandatory fields as Optional so they could be found at a glance?

I guess only `modelName` is required, all other fields are optional. But for `modelName` the "required" is conditional, depending where/when it is specified (to the point above).

> Slightly related to this should we return some default value when such a meaningful default exists? For instance should the temperature() return 0.0 by default or the maxOutputTokens() return Integer.MAX_VALUE?

Hmm, I am not sure we should be opinionated on the defaults here:
- different use cases require different setting
- a specific default value in one model might not make the same effect in another model
- most of the parameters are optional and default values are applied on the LLM API side

So I would rateher keep them `null` by default and let LLM API decide the default.